2-mandatory/4-stocks.js

```diff
@@ -64,31 +83,58 @@ function getPriceChanges(closingPricesForAllStocks) {
     The price should be shown with exactly 2 decimal places.
 */
 function highestPriceDescriptions(closingPricesForAllStocks, stocks) {
-    // TODO
-}
+  // TODO
+  let highestOfStock = [];
+  let stockName = [];
+  let describingArr = [];
+
+  for (let item of closingPricesForAllStocks) {
+    let maxPrice = 0;
+    for (let price of item) {
+      if (price > maxPrice) {
+        maxPrice = price;
+      }
+    }
+
+    highestOfStock.push(maxPrice.toFixed(2));
+    //highestOfStock.push(Math.max.apply(null,item));   --> OK
+    // Math.max(array) --> not supported in the current version (will return NaN)
+  }
 
+  for (let element of stocks) {
+    stockName.push(element.toUpperCase());
+  }
+
+  for (let i = 0; i < 5; i++) {
+    let sentence = `The highest price of ${stockName[i]} in the last 5 days was ${highestOfStock[i]}`;
+    describingArr.push(sentence);
+  }
+
+  return describingArr;
+}
```

Comment:
well done on solving this! you could reduce your amount of for loops, by just having one loop which loops through the array of `CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS` and then uses `stocks[i] `to select the relevant stock